sound/soc/intel/boards/sof_sdw.c

```diff
@@ -863,8 +863,10 @@ static int sof_card_dai_links_create(struct device *dev,
 	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
 		codec_info_list[i].amp_num = 0;
 
-	hdmi_num = sof_sdw_quirk & SOF_SDW_TGL_HDMI ?
-				SOF_TGL_HDMI_COUNT : SOF_PRE_TGL_HDMI_COUNT;
+	if (sof_sdw_quirk & SOF_SDW_TGL_HDMI)
+		hdmi_num = SOF_TGL_HDMI_COUNT;
+	else
+		SOF_PRE_TGL_HDMI_COUNT;
```

Comment:
this is wrong. In fact I'm wondering: you write "There's no reason to use the ternary operator here." What do you consider to be a valid reason for a ternary operator?

Reply:
Mark Brown doesn't want ternary operators, so we do what maintainers say.

Reply:
fixed